benchmarks/index.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+const { PassThrough } = require('stream')
+const http = require('http')
+const Benchmark = require('benchmark')
+const undici = require('..')
+
+const httpOptions = {
+  protocol: 'http:',
+  hostname: 'localhost',
+  method: 'GET',
+  path: '/',
+  port: 8080,
+  agent: new http.Agent({ keepAlive: true })
+}
+
+const undiciOptions = {
+  path: '/',
+  method: 'GET'
+}
+
+const pool = undici(`http://${httpOptions.hostname}:${httpOptions.port}`, {
+  connections: 100,
+  pipelining: 10
+})
+
+const suite = new Benchmark.Suite()
+
+suite.add('http - keepalive', {
+  defer: true,
+  fn: deferred => {
+    http.get(httpOptions, response => {
+      response.resume()
+      response.once('end', () => {
+        deferred.resolve()
+      })
+    })
+  }
+}).add('undici - request', {
+  defer: true,
+  fn: deferred => {
+    pool.request(undiciOptions, (error, { body }) => {
+      if (error) {
+        throw error
+      }
+
+      body.resume()
+      body.once('end', () => {
+        deferred.resolve()
+      })
```

Comment:
Can you pipe this to a pass through? Same as stream version

Reply:
You sure? If we do `pool.stream` we must provide a writable stream, right? But if you use the Node.js API then it automatically produces a stream. I mean to avoid unnecessary overhead.

Reply:
Otherwise the comparison with stream is unfair.

Reply:
But `undici` directly calls `body.write` (on the user provided stream) whereas Node.js has to generate the body by itself...

Reply:
So the point of `request.stream` is that `request.stream` can directly write to the final destination, while `stream.request` has to `pipe` it. However, if you don't `pipe` to something in the `stream.request` case then this advantage is not shown in the benchmark.

Reply:
I see your point now. I will update the tests.